### PR TITLE
FIXED: adapter deployment permission denied error on docker platform

### LINKED ISSUE #8693: [mesheryctl] Permission denied error while deploying adapters in docker environment
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

on Meshery Dashboard in Settings->Adapters when deploying an adapter getting below the `Permission denied` error in logs.
problem

and the user is unable to deploy and connect to adapters.
### Environment

- Host OS: Mac Linux Windows
- Meshery Server Version: stable-v
- Meshery Client Version: stable-v
- Platform: Docker Kubernetes

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- 📺 [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- ⌨️ [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- ⌨️ [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by zakisk**
**Notes for Reviewers**

This PR fixes #8693

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

Untitled design (1)

Untitled design

### CODE REVIEW

**Review by leecalcote — CHANGES_REQUESTED** [review]
@zakisk, please address failed lint checks.

@nebula-aac how does the rest of this PR look?

**Comment by zakisk** [comment]
> @zakisk, please address failed lint checks.
> 
> @nebula-aac how does the rest of this PR look?

@leecalcote sure!

File: mesheryctl/internal/cli/root/system/start.go [hunk 1]
```diff
@@ -303,6 +298,21 @@ func start() error {
...
+		}
+
+		// Create the group_add option and add GID of docker group to meshery container
+		groupAdd := viper.GetStringSlice("services.meshery.group_add")
+		groupAdd = append(groupAdd, group.Gid)
+		utils.ViperCompose.Set("services.meshery.group_add", groupAdd)
+
+		// Write the modified configuration back to the Docker Compose file
+		if err := utils.ViperCompose.WriteConfig(); err != nil {
+			return errors.Wrap(err, utils.SystemError("unable to add group_add option"))
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/start.go:313]:
```suggestion
			return errors.Wrap(err, utils.SystemError("unable to add group_add option. Meshery Server cannot perform this privileged action"))
```

**Comment by leecalcote** [comment]
@zakisk we're not getting all of Meshery Server's errors sent to the /apis/events, are we?

For instance, I see this error on stdout, but not via Meshery UI:

```
ERRO[2023-09-13T16:07:12-05:00] the platform local is not supported currently. The supported platforms are: Docker and Kubernetes  app=meshery code=1231 probable-cause="Your platform is not supported for deploying Meshery Adapters" severity=3 short-description="Unable to deploy Meshery Adapter in the current environment" suggested-remediation="Consider using a supported platform for deploying Meshery Adapters"
```

I wonder if you can help with this in a separate PR.

@MUzairS15 guidance?

**Comment by zakisk** [comment]
> @zakisk we're not getting all of Meshery Server's errors sent to the /apis/events, are we?
> 
> For instance, I see this error on stdout, but not via Meshery UI:
> 
> ```
> ERRO[2023-09-13T16:07:12-05:00] the platform local is not supported currently. The supported platforms are: Docker and Kubernetes  app=meshery code=1231 probable-cause="Your platform is not supported for deploying Meshery Adapters" severity=3 short-description="Unable to deploy Meshery Adapter in the current environment" suggested-remediation="Consider using a supported platform for deploying Meshery Adapters"
> ```
> 
> I wonder if you can help with this in a separate PR.
> 
> @MUzairS15 guidance?

will check and let you know.

**Comment by leecalcote** [comment]
@zakisk will you open an issue? If not, @MUzairS15 will you carry this forward?

**Pull request merged**